src/gluonts/dataset/loader.py

```diff
@@ -11,186 +11,270 @@
 # express or implied. See the License for the specific language governing
 # permissions and limitations under the License.
 
-# Standard library imports
+from typing import Iterable, Iterator, Callable, Optional
+import itertools
+import random
 import logging
-from typing import Iterable, Iterator, Optional
+from multiprocessing import Process, Manager, Queue
+from multiprocessing.reduction import ForkingPickler
+import io
+import pickle
+import sys
+from queue import Empty
 
-# Third-party imports
-import mxnet as mx
-import numpy as np
-
-# First-party imports
-from gluonts.core.component import DType
-from gluonts.dataset.common import DataBatch, DataEntry, Dataset
-from gluonts.dataset.parallelized_loader import ParallelDataLoader
+from gluonts.dataset.common import DataEntry, DataBatch, Dataset
+from gluonts.dataset.util import MPWorkerInfo, batcher
 from gluonts.transform import Transformation
+from gluonts.transform.dataset import TransformedDataset
+
+
+logger = logging.getLogger(__name__)
 
 
-class DataLoader(Iterable[DataEntry]):
+class CyclicIterable(Iterable):
```

Comment:
```suggestion
class Cycle(Iterable):
```

I think it makes sense to abbreviate this. We are not Java after all.
`itertools.cycle`